lib/segment/src/spaces/metric_f16/simple_avx.rs

```diff
@@ -0,0 +1,319 @@
+use std::arch::x86_64::*;
+
+use common::types::ScoreType;
+use half::f16;
+
+use crate::data_types::vectors::VectorElementTypeHalf;
+
+#[target_feature(enable = "avx")]
+#[target_feature(enable = "fma")]
+#[allow(clippy::missing_safety_doc)]
+pub unsafe fn hsum256_ps_avx(x: __m256) -> f32 {
+    let x128: __m128 = _mm_add_ps(_mm256_extractf128_ps(x, 1), _mm256_castps256_ps128(x));
+    let x64: __m128 = _mm_add_ps(x128, _mm_movehl_ps(x128, x128));
+    let x32: __m128 = _mm_add_ss(x64, _mm_shuffle_ps(x64, x64, 0x55));
+    _mm_cvtss_f32(x32)
+}
+
+#[target_feature(enable = "avx")]
+#[target_feature(enable = "fma")]
+#[target_feature(enable = "f16c")]
+pub(crate) unsafe fn euclid_similarity_avx(
```

Comment:
One comment for all namings. `euclid_similarity_avx` is already presented. Please, rename `euclid_similarity_avx` into `avx_euclid_similarity_half` like it was named for byte type:
https://github.com/qdrant/qdrant/blob/dev/lib/segment/src/spaces/metric_uint/avx2/euclid.rs#L9

Reply:
Do this please for all simd functions

Reply:
Done